### PR TITLE
Girders and wall construction

### LINKED ISSUE #1726: Means to build walls ingame B: 60
# Description
At the moment we can only remove walls by welding them. Need ability to place them back!

### PR DESCRIPTION
**Pull request by Aranclanos**
### Purpose
Fixes #1726
Adds girders, players will now be able to build them using metal on their hand, and finish the process of wall building by applying more metal on the girder.

Reduces the amount of metal sheets spawned by disassembling walls to two.
Adds the structures png with only the sprite girder created.

Moves DisappearObject() from Grenade code to InputTrigger and PickUpTrigger.
Adds a new prefab for girders.
Adds MetalTrigger.cs and GirderTrigger.cs

image

### Open Questions and Pre-Merge TODOs

- [X]  This fix is tested on the same branch it is PR'ed to.
- [X]  I correctly commented my code
- [X]  My code is indented with tabs and not spaces
- [X]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [X]  This PR does not bring up any new compile errors
- [X]  This PR has been tested in editor
- [x]  This PR has been tested in multiplayer (with 2 clients, if applicable)

### Notes:
the girders only work serverside only, CanUse() is broken in develop at the moment

### CODE REVIEW

**Comment by chairbender** [comment]
Re: CanUse being broken (sorry for the spam but posting this here as well for visibility)

@Aranclanos

I wouldn't call it broken. The only change I made was that it no longer has the side effect of sending an interaction message to the server when it is invoked on the client side. I removed this behavior because it doesn't make sense to have a method called "CanUse" which has a quite significant and undocumented side effect of performing network communication - the method was doing too much by having this side effect. CanUse should only be used to check if something is usable.

Please refactor your code to send this message in a different way.

In the meantime I will work on the other users of this method to ensure they do the same thing.

EDIT: For reference, the logic that was removed was from InputController.CanUse:
if (!isServer)
{
  InteractMessage.Send(gameObject, position, hand);
}

**Comment by chairbender** [comment]
FYI @Aranclanos I just merged a fix for this problem for all the other users of CanUse. Take a look at that if you're unsure how to send the message now. PR was #1759

**Comment by Aranclanos** [comment]
Done

**Comment by chairbender** [comment]
Wow, really nice! Tested it out and it seems fine. Going over the code now...

File: UnityProject/Assets/Scripts/Items/MetalTrigger.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Networking;
+
+public class MetalTrigger : PickUpTrigger
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/Items/MetalTrigger.cs:9]: please add /// comment describing the component

File: UnityProject/Assets/Scripts/Objects/GirderTrigger.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections;
+using UnityEngine;
+using UnityEngine.Networking;
+using Random = UnityEngine.Random;
+
+public class GirderTrigger : InputTrigger
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/Objects/GirderTrigger.cs:9]: please add /// comment describing the component

**Review by chairbender — CHANGES_REQUESTED** [review]
just want to get some comments describing those new classes, other than that looks good

**Comment by DooblyNoobly** [comment]
Let me know once the comments are added and I'll resolve these conflicts

**Comment by Aranclanos** [comment]
Done

**Comment by DooblyNoobly** [comment]
thanks for fixing the conflicts too!

**Pull request merged**